### PR TITLE
Parallel integrator plots

### PR DESCRIPTION
**Pull request by eggrobin**
We had 18 500 independent problems for each set of graphs (position-velocity-energy), so this falls under "embarrassingly parallel".

### CODE REVIEW

File: mathematica/integrator_plots.cpp [hunk 1]
```diff
@@ -169,15 +172,148 @@ std::vector<SimpleHarmonicMotionPlottedIntegrator> ReferenceMethods() {
...
+    std::string result;
+    result += Assign("qErrorData", q_error_data);
+    result += Assign("vErrorData", v_error_data);
+    result += Assign("eErrorData", e_error_data);
+    result += Assign("names", names);
+    return result;
+  }
+
+ private:
+  Status Integrate(int method_index, int time_step_index) {
```
Comment (pleroy) [inline mathematica/integrator_plots.cpp]: const (2x)

File: mathematica/integrator_plots.cpp [hunk 2]
```diff
@@ -169,15 +172,148 @@ std::vector<SimpleHarmonicMotionPlottedIntegrator> ReferenceMethods() {
...
+    evaluations_.resize(methods_.size());
+    for (int i = 0; i < methods_.size(); ++i) {
+      q_errors_[i].resize(integrations_per_integrator_);
+      v_errors_[i].resize(integrations_per_integrator_);
+      e_errors_[i].resize(integrations_per_integrator_);
+      evaluations_[i].resize(integrations_per_integrator_);
+    }
+  }
+
+  std::string GetData() {
```
Comment (pleroy) [inline mathematica/integrator_plots.cpp]: Say "Mathematica".

File: mathematica/integrator_plots.cpp [hunk 3]
```diff
@@ -169,15 +172,148 @@ std::vector<SimpleHarmonicMotionPlottedIntegrator> ReferenceMethods() {
...
+    evaluations_[method_index][time_step_index] = amortized_evaluations;
+
+    return Status::OK;
+  }
+
+  std::vector<SimpleHarmonicMotionPlottedIntegrator> const methods_;
+  std::function<void(Instant const& t,
+                     std::vector<Length> const& q,
+                     std::vector<Acceleration>& result,
+                     int* evaluations)>
```
Comment (pleroy) [inline mathematica/integrator_plots.cpp]: `int&`?

File: mathematica/integrator_plots.cpp [hunk 4]
```diff
@@ -169,15 +172,148 @@ std::vector<SimpleHarmonicMotionPlottedIntegrator> ReferenceMethods() {
...
+    auto append_state = [this, &max_q_error, &max_v_error, &max_e_error](
+      ODE::SystemState const& state) {
+      auto const errors = compute_errors_(state);
+      max_q_error = std::max(max_q_error, errors.q_error);
+      max_v_error = std::max(max_v_error, errors.v_error);
+      max_e_error = std::max(max_e_error, errors.e_error);
+    };
+    Time const Δt = method.evaluations * starting_step_size_per_evaluation_ /
+                    std::pow(step_reduction_, time_step_index);
+    auto instance = method.integrator.NewInstance(problem, append_state, Δt);
```
Comment (pleroy) [inline mathematica/integrator_plots.cpp]: auto const

File: mathematica/integrator_plots.cpp (hunk 2 above)
Comment (eggrobin) [inline mathematica/integrator_plots.cpp]: Done.

File: mathematica/integrator_plots.cpp (hunk 1 above)
Comment (eggrobin) [inline mathematica/integrator_plots.cpp]: Done.

File: mathematica/integrator_plots.cpp (hunk 3 above)
Comment (eggrobin) [inline mathematica/integrator_plots.cpp]: No, it's optional.

File: mathematica/integrator_plots.cpp (hunk 4 above)
Comment (eggrobin) [inline mathematica/integrator_plots.cpp]: Done.

File: mathematica/integrator_plots.cpp (hunk 3 above)
Comment (pleroy) [inline mathematica/integrator_plots.cpp]: Then it should really be `std::optional<int>& evaluations`.  Please put a TODO in `testing_utilities`.

Comment (eggrobin) [inline mathematica/integrator_plots.cpp]: No, as discussed

**Pull request merged**